### PR TITLE
URL prameter encoding/decoding added

### PR DESCRIPTION
**Pull request by a-breid-ericsson**
Some URL parameters could not be handles as JMESPATH may include invalid characters. This is now handled better so that the JMESPATH expression is not needed to be urlencoded in the notification meta.
The url is encoded before sending it to recieving part although sonce it is impossible to use it if not and if needed.

URL parameters may both be encoded or uncoded since EI will decode if needed the parameters before sending them through the JMESPATH process.

Validation that validated that notification meta was valid has been changed since it declined JMESPATH expressions, now it only validates that it is not empty, and that it is a valid email address if it is a mail notification type. 

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:

### CODE REVIEW

File: src/main/java/com/ericsson/ei/subscriptionhandler/SubscriptionValidator.java [hunk 1]
```diff
@@ -121,24 +121,34 @@ private static void validateNotificationMessageKeyValues(List<NotificationMessag
 
     /**
      * Validation of notificationMeta parameter Throws
-     * SubscriptionValidationException if validation of the parameter fails due to
-     * wrong format of parameter.
+     * SubscriptionValidationException if validation of the parameter fails due
```
Comment (Christoffer-Cortes) [inline src/main/java/com/ericsson/ei/subscriptionhandler/SubscriptionValidator.java]: Maybe the Javadoc description should change a little now that the function has changed?

File: src/test/java/com/ericsson/ei/subscriptionhandler/test/SubscriptionValidatorTest.java [hunk 2]
```diff
@@ -356,28 +356,54 @@ public void validateRestPostMediaTypeInvalidMessage2Test() throws Exception {
...
+
+    @Test
+    public void validateNotificationMetaInvalidMetaTest() throws Exception {
+        String notificationMeta = "";
+        String notificationType = "REST_POST";
         try {
-            invokeMethod(subscriptionValidator, "validateNotificationMeta", notificationMeta);
+            invokeMethod(subscriptionValidator, "validateNotificationMeta", notificationMeta, notificationType);
         } catch (SubscriptionValidationException e) {
             assertTrue(e.getMessage(), true);
```
Comment (evasiba-ericsson) [inline src/test/java/com/ericsson/ei/subscriptionhandler/test/SubscriptionValidatorTest.java]: please remove all assertTrue(...,True) in this file even if they are not written by you

File: src/main/java/com/ericsson/ei/subscriptionhandler/SubscriptionValidator.java [hunk 3]
```diff
@@ -121,24 +121,34 @@ private static void validateNotificationMessageKeyValues(List<NotificationMessag
...
-        } else if (Pattern.matches(regex, notificationMeta)) {
-            throw new SubscriptionValidationException("Wrong format of NotificationMeta: " + notificationMeta);
+        }
+
+        if (Pattern.matches(regexMail, notificationType)) {
+            String regexEmailCheck = "^[\\w-\\+]+(\\.[\\w]+)*@[\\w-]+(\\.[\\w]+)*(\\.[a-zA-Z]{2,})$";
+            boolean isInvalidEmailAddress = !Pattern.matches(regexEmailCheck, notificationMeta);
+            if (isInvalidEmailAddress) {
+                throw new SubscriptionValidationException(
+                        "Wrong format of NotificationMeta on the selected type.\nNotification meta: " + notificationMeta
```
[the remark this answers is not in the document]
Comment (a-breid-ericsson) [reply src/main/java/com/ericsson/ei/subscriptionhandler/SubscriptionValidator.java]: Done

File: src/functionaltests/resources/features/subscriptionCRUD.feature [hunk 4]
```diff
@@ -22,28 +22,31 @@ Feature: Test Subscription CRUD
...
+    When I make a PUT request with modified "<key>" as "<newValue>" to REST API "/subscriptions"
     Then I get response code of 200
-    And  I can validate modified notificationType "MAIL" with GET request at "/subscriptions/Subscription_Test"
-  
+    And  I can validate modified "<key>" "<newValue>" with GET request at "/subscriptions/Subscription_Test"
+
+# Values to be added to the variables given above
+  Examples:
+    | key              | newValue                         |
+    | notificationMeta | new.notification.meta/some/path/ |
```
Comment (e-pettersson-ericsson) [inline src/functionaltests/resources/features/subscriptionCRUD.feature]: If only one set of example values are tested, then I don't see the point of having a scenario outline with examples. It's easier to read which parameters are sent in to the AND statement if it's written like before.

Comment (a-breid-ericsson) [reply src/functionaltests/resources/features/subscriptionCRUD.feature]: I see it as it is easier to change what parameter to change when updating. The save "value" and "key" is used on multiple places thus, you need to update more values if you need to change them.

Comment (a-breid-ericsson) [reply src/functionaltests/resources/features/subscriptionCRUD.feature]: I wish cucumber had better ways to add reused variables for the whole file.

**Pull request merged**